torch/csrc/jit/codegen/cuda/kernel.cpp

```diff
@@ -300,6 +301,73 @@ void validateKernelArgs(
         msg.str());
   }
 }
+
+size_t size(const dim3& d) {
```

Comment:
This `size` function is just a utility function I often use. It's used for computing the size of allocations but also for other purposes like computing offsets in shared and global memories. It's just a pure utility function.

I do see the point of having global memory allocations as IR nodes. That could be a good approach to handle memory allocations within the same overall framework, rather than handling individual needs of memories in an ad-hoc, intertwined way like this.